torchvision/utils.py

```diff
@@ -210,3 +210,59 @@ def draw_bounding_boxes(
             draw.text((bbox[0], bbox[1]), labels[i], fill=color, font=txt_font)
 
     return torch.from_numpy(np.array(img_to_draw)).permute(2, 0, 1)
+
+
+@torch.no_grad()
+def draw_segmentation_masks(
+    image: torch.Tensor,
+    masks: torch.Tensor,
+    colors: Optional[List[Union[str, Tuple[int, int, int]]]] = None,
+) -> torch.Tensor:
+
+    """
+    Draws segmentation masks on given image.
+    The values of the input image should be uint8 between 0 and 255.
+
+    Args:
+        image (Tensor): Tensor of shape (C x H x W)
+        masks (Tensor): Tensor of shape (H, W). Each containing predicted class.
```

Comment:
Have you considered passing the raw, unmodified mask output here (before applying `.argmax(0)`). This would make it aligned with the `draw_bounding_boxes()` util and also it will give you the ability to retrieve the number of classes.